2020-01-Winter/4_trees/README.md

```diff
@@ -116,21 +145,61 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s2"/>
 
-### 2. SOLUTION 2 TODO :bug:
-
-Source: TODO :bug:
+### 2. SOLUTION 2
+
+Source:
+
+#### Recursion Solution
```

Comment:
For each block of solution code that gets added, I'd like to see at least a sentence or two that gives an overview of the solution's strategy.

In this case, we would say that we set up a `counter` variable, initialized to `k`. We then traverse the tree using **in-order traversal**, and with each visited node, we decrement `counter` by 1. When `counter` is 0, we have found the value we are looking for.

Can you come up with a similar overview for the iterative solution?